doc/Coding_guidelines.md

```diff
@@ -191,9 +191,11 @@ Feel free to improve this document. Bring more examples and rules. But please tr
 
 10. Use unsigned types whenever it possible. Unsigned overflow is well defined behavior while signed is not.
 
+11. Avoid usage of ```static``` variables because it might bring unexpectable overhead. Since C++11 initialization of ```static``` variables must be thread safe, so compiler generates additional code to provide it.
```

Comment:
I'd like to note that it is so only for [Static local variables](https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables).

Reply:
I will reflect this in description